kurtosis-devnet/pkg/kurtosis/sources/deployer/deployer.go

```diff
@@ -236,11 +236,11 @@ func parseStateFile(r io.Reader) (*DeployerState, error) {
 	}
 
 	mapDeployment := func(deployment map[string]interface{}) DeploymentAddresses {
-		addrSuffix := "Address"
+		addrSuffix := "Proxy"
 		addresses := make(DeploymentAddresses)
 		for key, value := range deployment {
 			if strings.HasSuffix(key, addrSuffix) {
-				addresses[strings.TrimSuffix(key, addrSuffix)] = common.HexToAddress(value.(string))
+				addresses[key] = common.HexToAddress(value.(string))
 			}
 		}
 		return addresses
```

Comment:
Some addresses have the suffix `Impl` instead of `Proxy`. As this code is written right now, it will ignore all `Impl` addresses. So maybe we should check for both suffixes now?
```
        addresses := make(DeploymentAddresses)
		for key, value := range deployment {
                 if strings.HasSuffix(key, "Proxy") || strings.HasSuffix(key, "Impl") {
				    addresses[key] = common.HexToAddress(value.(string))
                 }
        }
```
Or maybe just use `deployment` directly without having to convert to a `addresses` map?

Reply:
Thanks, I wasn't sure if anybody wanted the Impl addresses. I guess it won't hurt to add them.

I can't use `deployment` directly as it's a non-homogenous map, with non-addresses in there

Reply:
Part of my pr that triggered this whole journey actually uses a embedded homogenous struct to contain the addresses instead of a flat struct with mixed types ([ref](https://github.com/ethereum-optimism/optimism/blob/develop/op-deployer/pkg/deployer/state/state.go#L92-L102)).

So you should just be able to pull `state.OpChainAddresses` and get all the addresses that way. Doesn't have to block this pr but those are the kind of changes I'm trying to make as part of the address standardization effort